pandas/_libs/internals.pyi

```diff
@@ -44,7 +44,9 @@ class BlockPlacement:
     @property
     def is_slice_like(self) -> bool: ...
     @overload
-    def __getitem__(self, loc: slice | Sequence[int]) -> BlockPlacement: ...
+    def __getitem__(
```

Comment:
can the ndarray be made more specifically np.intp?  in particular im thinking ruling out bool

Reply:
Yep, updated